faq.md

```diff
@@ -575,6 +575,144 @@ If you are using the Flutter plugin for IntelliJ, you can launch the app in
+subclasses. For example, our render objects have a base class
+([RenderObject](https://docs.flutter.io/flutter/rendering/RenderObject-class.html))
+that is agnostic regarding the coordinate system, and then we have a subclass
+([RenderBox](https://docs.flutter.io/flutter/rendering/RenderBox-class.html))
+that introduces the opinion that the geometry should be based on the Cartesian
+coordinate system (x/width and y/height).
+
+* Prototype-based object-orientated programming: The
+[ScrollPhysics](https://docs.flutter.io/flutter/widgets/ScrollPhysics-class.html)
+class chains instances to compose the physics that apply to scrolling
+dynamically at runtime. This lets the system compose, for example, paging
+physics with platform-specific physics, without the platform having to be
+selected at compile time.
+
+* Imperative programming: Straightforward imperative programming, usually
+paired with state encapsulated within an object, is used where it provides the
+most intuitive solution. For example, tests are written in an imperative style,
+first describing the situation under test, then listing the invariants that the
+test must match, then advancing the clock or inserting events as necessary for
+the test.
+
+* Reactive programming: The widget and element trees are sometimes described as
+reactive, because new inputs provided in a widget's constructor are immediately
+propagated as changes to lower-level widgets by the widget's build method, and
+changes made in the lower widgets (e.g. in response to user input) propagate
+back up the tree via event handlers. Aspects of both functional-reactive and
+imperative-reactive are present in the framework, depending on the needs of the
+widgets. For example, the material
+[Divider](https://docs.flutter.io/flutter/material/Divider-class.html) class has
+a build method that consists of just an expression describing how this widget
+reacts to changes in its configuration (a functional reactive widget), while the
+[Chip](https://docs.flutter.io/flutter/material/Chip-class.html) class has a
+build method that constructs a list of children over several statements,
+describing how this widget reacts to changes in its configuration (an imperative
+reactive widget).
+
+* Declarative programming: The build methods of widgets are often a single
+expression with multiple levels of nested constructors, written using a strictly
+declarative subset of Dart (such nested expressions could be mechanically
+transformed to or from any suitably expressive markup language). For example,
+the
+[UserAccountsDrawerHeader](https://docs.flutter.io/flutter/material/UserAccountsDrawerHeader-class.html)
+widget has a long build method (20+ lines), consisting of a single nested
+expression. This can also be combined with the imperative style to build UIs
+that would be harder to describe in a pure-declarative approach.
+
+* Generic programming: Types can be used to help catch programming errors early.
```

Comment:
delete "help"

Reply:
Why? The types themselves don't catch any errors. It's the programmer who catches the errors with the help of the analyzer which itself is helped by the type declarations in generics.

Reply:
If that is true, then why are you leaving out the subject (who/what is catching the errors)?

Reply:
Ah, I see. Added "developers".